### ISSUE TITLE
MiniWorldReference Appears in Scene 

### ISSUE THREAD

**Issue by sarahhockman** [Unity-Technologies/EditorVR#86]
[**PR #78**, Using EVR-staging, 9 March 2017], using the VIVE + Oculus

When viewing the MiniWorld, I select the player (MiniWorldReference) with the trigger, and attempt to move it to another point in the scene, the result is the MiniWorldReference appearing outside the MiniWorld and in the physical scene. It appears in the Hierarchy as MiniWorldReference, and can even be selected, manipulated, and deleted. When deleted, or if the MiniWorld is closed, I am left with a sphere where MiniWorldReference was previously in the scene, which cannot be selected or deleted. (Can be seen in gif below). 
![ezgif com-optimize](https://cloud.githubusercontent.com/assets/25187040/23776274/1b7d3e7c-04e2-11e7-81d0-2a4b0d63d431.gif)

- **Note: In the HMD, you can see the sphere and rectangle left behind [MiniWorldReference], but they did not appear on my monitor’s device view.**   

When re-opening the MiniWorld and zooming in on the player, MiniWorldReference no longer appears, but the [VIVE/Touch] controllers animate as I move them.
 
_Replicate this instance by:_
1. Open MiniWorld
2. Select MiniWorldReference (player)
3. Drag to another point in scene
4. Sphere+Rectangle (player shape) appears in scene

_Via Hierarchy: “MiniWorldReference”, two objects appear (rectangle and sphere) can select the rectangle, sphere is not selectable. In Device View cannot see objects but gizmo appears, in HMD can see objects + gizmo_

**Comment by sarahhockman** [comment]
[[ Error, cannot approve ]]

**Using EditorVR-staging, pulled on 14 March 2017:**

1) I can confirm that one part of this issue is resolved. When using the MiniWorld and grabbing the player, MiniWorldReference no longer appears in the Hierarchy workspace.

However, 

2) I am unable to move the player view (myself) in the MiniWorld. When I attempt to grab the player view, the MiniWorldReference still appears in my physical scene.

_**Note: In the HMD, I can see the sphere and box shape (MiniWorldReference) in my physical scene, but it does not appear in my monitor device view. Also, although it appears in my physical scene (and can be selected/manipulated), it is hidden in the Hierarchy.**_

The player view freezes in the MiniWorld workspace (The player view can be highlighted, but cannot be moved in the MiniWorld). Instead, in my physical scene, a box (from MiniWorldReference) appears, and is attached to my player view in the MiniWorld. This box can be selected and manipulated, and I can see that it is attached to the player view [MiniWorldReference] in the MiniWorld workspace.  

To summarize: 
- The MiniWorldReference appears in my physical scene when I attempt to select the player view and move it in the MiniWorld workspace.
- In my physical scene (in the HMD), I can see a sphere and box shape left behind by the above instance.
- In my physical scene, I cannot select the sphere, but I am able to select the box.
- When I refer back to the player view in the MiniWorld, I can still highlight the MiniWorldReference, but I am only able to select it from the physical scene. When I do select it, the sphere remains frozen, while the box can be moved and manipulated.

I'll attempt to illustrate this instance in the gifs below:

![ezgif com-optimize 1](https://cloud.githubusercontent.com/assets/25187040/23921107/9b92079c-08ba-11e7-8737-1160d60ba21e.gif)
_In the above gif, I have the box selected (which does not appear on my monitor), but what you can see is the free movement sphere of the manipulator gizmo, which is how I am moving the MiniWorldReference._


![ezgif com-crop 6](https://cloud.githubusercontent.com/assets/25187040/23921634/90d141e0-08bc-11e7-882e-cd0561e66862.gif)
_From the MiniWorld workspace: Box and Sphere are jointly highlighted when traced with cursor, but only able to select and move the box._


![ezgif com-resize](https://cloud.githubusercontent.com/assets/25187040/23921269/3aa19f64-08bb-11e7-9fd0-c6910b6fbbbd.gif)
![ezgif com-crop 5](https://cloud.githubusercontent.com/assets/25187040/23921271/4118b102-08bb-11e7-8bdf-23932a48f792.gif)
_Hard to tell because it is so small -- but in the MiniWorld, what I am moving is the box left behind by MiniWorldReference in the physical scene. The sphere of the player view remains frozen._

**Comment by sarahhockman** [comment]
Using staging as of 3/15 as of 2:20PM

_git.exe checkout staging --
Your branch is up-to-date with 'origin/staging'.
Switched to branch 'staging'
Success (63 ms @ 3/15/2017 2:21:40 PM)_

Apologies for the quality, pics taken from Oculus Lenses to show what is being seen within headset:

![img_6434](https://cloud.githubusercontent.com/assets/25187040/23972147/ed9544ea-098d-11e7-9bb4-7c8987724a5a.JPG)
1. _MiniWorld player head still appears in scene after selected and moved (did not disappear after de-selected, nor did it change my location in the scene)._

![img_6435](https://cloud.githubusercontent.com/assets/25187040/23972150/ed96603c-098d-11e7-91b4-333f1138fd8c.JPG)
2. Still appears in MiniWorld, can be highlighted but not moved.

![img_6436](https://cloud.githubusercontent.com/assets/25187040/23972148/ed95940e-098d-11e7-86c7-1d876cf26a8c.JPG)
3. _In physical scene, can be selected, gizmo appears. (Gizmo draws over the box, not the player head/sphere)._

![img_6439](https://cloud.githubusercontent.com/assets/25187040/23972149/ed95b6be-098d-11e7-8525-7c95387f1bfb.JPG)
4. _If I locomote in my scene, the MiniWorld player head remains where it was when it originally appeared in the physical scene. All I can see that matches myself are my controllers (as seen above, touch controllers move independently from sphere, appear where I am located in physical scene)._

**Comment by mtschoen-unity** [comment]
Hey, @sarahhockman, thanks for following up. We actually made this change to the development branch, but it didn't make it back into staging. When we finish this sprint, we will update staging, and start a second round of testing.  If you still would like to verify the fix, check out development and then try it.  I wouldn't worry about it, honestly. I'm fairly confident that we have resolved the issue. We can test this one last time before releasing 0.0.3
